### ISSUE TITLE
Address obsolete ContentItemGroup.FindItems in PackageValidation

### ISSUE THREAD

**Issue by ericstj** [dotnet/sdk#23301]
We use `ContentItemCollection.FindItemGroups` in package validation to replicate NuGet's asset selection algorithm:
https://github.com/dotnet/sdk/blob/c63af091c2e14a275bca031f00309ba9fe1f6eb1/src/Compatibility/Microsoft.DotNet.PackageValidation/Package.cs#L33-L39

As of https://github.com/NuGet/NuGet.Client/commit/bcd6856405ebaac80ea63819d0b5a7d87784eae5 this method is obsolete.  The newly introduced method `PopulateItemGroups` does too much (allocating groups that we'd have to flatten).

For now I've disabled the obsoletion warning.  We should decide what to do moving forward, either have NuGet add an API or change how we do asset selection in the package.  It may be convenient to solve this while addressing https://github.com/dotnet/sdk/issues/17364

Originally found here: https://github.com/dotnet/sdk/pull/23277#issuecomment-1006172039

**Comment by ericstj** [comment]
cc @jebriede @safern

**Comment by jebriede** [comment]
@ericstj thanks for tagging me on this issue. FindItems was cleaned up and marked as obsolete as part of a memory overallocation fix but looking at this more closely, I believe FindItems should remain and not be obsolete.

PopulateItemGroups replaces FindItemGroups which uses yield returns, and on a hot code path, resulted in the overallocation of enumerator objects. FindItems used to use yield returns as well but I fixed its implementation to no longer use yield returns so it does not need to be obsolete.

I've created a bug https://github.com/NuGet/Home/issues/11487 and submitted a fix to remove the obsolete attribute from FindItems in this PR: https://github.com/NuGet/NuGet.Client/pull/4394. Thanks!
